authors/apps/profiles/views.py

```diff
@@ -0,0 +1,57 @@
+
+from rest_framework import status
+from rest_framework.generics import RetrieveAPIView, ListAPIView, CreateAPIView, RetrieveUpdateAPIView
+from rest_framework.permissions import(
+    AllowAny,IsAuthenticatedOrReadOnly, IsAuthenticated
+)
+from .permissions import  IsOwnerOrReadOnly
+from django.shortcuts import get_object_or_404
+from rest_framework.response import Response
+from .models import Profile
+from .serializers import ProfileSerializer
+from .renderers import ProfileJSONRenderer
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import status
+
+
+class ProfileCreateAPIView(ListAPIView):
+    """
+    class to create a profile
+    """
+    permission_classes = (IsAuthenticated, )
+    queryset = Profile.objects.all()
+    serializer_class = ProfileSerializer
+
+    def list(self,request, format=None):
+        serializer = self.serializer_class(self.get_queryset(), many=True)
+        return Response(data=dict(profiles=serializer.data), status=status.HTTP_200_OK)
+
+
+class ProfileDetailAPIView(RetrieveUpdateAPIView):
+    permission_classes = (AllowAny,)
+    queryset = Profile.objects.all()
+    renderer_classes = (ProfileJSONRenderer, )
+    serializer_class = ProfileSerializer
+    lookup_field = 'username'
+
+    def get_object(self):
+        return get_object_or_404(
+            self.get_queryset(), user__username=self.kwargs.get('username')
+        )
+
+    def put(self, request, *args, **kwargs):
+        profile = self.get_object()
+        profile = Profile.objects.get(username=kwargs["username"])
```

Comment:
@felixkiryowa why do you have repeated definition of profile i.e. on line `43 and 44`